[Infra] Identify hosts not monitored by the system integration in the hosts list

jennypavlova opened the pull request:
Closes [#190418](https://github.com/elastic/kibana/issues/190418)

## Summary

This PR adds a popover for the hosts that are not monitored by the system integration to help users troubleshoot the problem

## Testing 

A way to check it is to have a host that is monitored by system integration (I am using metricbeat) and sythtrace hosts - there we should have both (I am using `node scripts/synthtrace infra_hosts_with_apm_hosts --scenarioOpts.numInstances=50 --live`)

- Go to hosts view and check the table
- The hosts that are **not** monitored by the system integration should have an icon before the host name which opens the popover with some explanation and documentation links
- The hosts that are monitored by the system integration should not have an icon before the host name 

<img width="961" alt="Screenshot 2024-08-22 at 12 02 27" src="https://github.com/user-attachments/assets/a3330f3f-33df-41ae-8135-71aea98e2660">

![image](https://github.com/user-attachments/assets/b28474d4-2272-4372-b70d-6370aa8e2a9d)

jennypavlova committed:
Add tooltip for apm hosts

jennypavlova committed:
Merge branch 'main' into 190418-infra-identify-hosts-not-monitored-by-the-system-integration-in-the-hosts-list

jennypavlova commented:
/ci

jennypavlova commented:
/ci

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

cauemarcondes reviewed (approved):
LGTM

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx:
```diff
@@ -267,6 +277,63 @@ export const useHostsTable = () => {
+                              <EuiLink
+                                data-test-subj="hostsViewTooltipDocumentationLink"
+                                href=" https://www.elastic.co/guide/en/observability/current/apm.html"
+                                target="_blank"
```
Good point - I will remove that, it doesn't make sense in this case

roshan-elastic replied:
Hey @jennypavlova - I'm not sure the EUI standard on this but when I speak to users, users get confused if an documentation link doesn't flag it is an external link and doesn't open in a new tab).

I would keep it external...

jennypavlova replied:
Thanks, I made it external and updated the screenshot in the [description](https://github.com/elastic/kibana/pull/191063#issue-2480381669)

jennypavlova committed:
Fix test and link

crespocarlos reviewed (commented):
Thanks for this PR! The ticket was not clear enough, but the APM link needs to display the external link icon, since it does it in a new tab.

<img width="331" alt="image" src="https://github.com/user-attachments/assets/e8f90e8f-1fb1-4c7e-ba4e-2725ce690b8b">

jennypavlova committed:
Make the apm link external

jennypavlova committed:
Merge branch 'main' into 190418-infra-identify-hosts-not-monitored-by-the-system-integration-in-the-hosts-list

jennypavlova commented:
> Thanks for this PR! The ticket was not clear enough, but the APM link needs to display the external link icon, since it does it in a new tab.

@crespocarlos Thanks for clarifying! I changed it back to external and updated the description

crespocarlos reviewed (approved):
LGTM!
